modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -279,148 +286,258 @@ public function createOrderTableIfNotExist() {
+
+        return bin2hex($cipherText);
+    }
 
-		if($existing_order) {
-			return false;
-		}
+	/*
+	 * Add a new skeleton order in the db
+	 */
+    public function getOrderHash($id_order, $amount, $currency)
+    {
+        return $this->encryptHash($id_order . ":" . $amount . ":" . $currency);
+    }
 
+	/*
+	 * Get all orders linked to id
+	 */	
+	public function getAllOrdersById($order_id) {
 		try {
-			Capsule::table('blockonomics_bitcoin_orders')->insert(
-				[
-					'id_order' => $id_order,
-					'addr' => $address,
-					'timestamp' => time(),
-					'status' => -1,
-					'value' => $value,
-					'bits' => $bits,
-				]
-			);
-		} catch (\Exception $e) {
-				echo "Unable to insert new order into blockonomics_bitcoin_orders: {$e->getMessage()}";
+			return Capsule::table('blockonomics_bitcoin_orders')
+				->where('id_order', $order_id)
+				->orderBy('timestamp', 'desc')->get();
+		} catch (Exception $e) {
+				exit("Unable to get orders from blockonomics_bitcoin_orders: {$e->getMessage()}");
 		}
-
-		return true;
 	}
 
 	/*
-	 * Try to get order row from db by address
-	 */
-	public function getOrderByAddress($bitcoinAddress) {
-		try {
-			$existing_order = Capsule::table('blockonomics_bitcoin_orders')
-				->where('addr', $bitcoinAddress)
-				->orderBy('timestamp', 'desc')
-				->first();
-		} catch (\Exception $e) {
-				echo "Unable to select order from blockonomics_bitcoin_orders: {$e->getMessage()}";
-		}
+	 * Check for pending orders and return if exists
+	 */	
+    public function getPendingOrder($orders)
+    {
+        foreach ($orders as $order) {
+            //check if status 0 or 1
+            if ($order->status == 0 || $order->status == 1) {
+                return $order;
+            }
+        }
+        return false;
+    }
 
-		$row_in_array = array(
-			"id" => $existing_order->id,
-			"order_id" => $existing_order->id_order,
-			"timestamp"=> $existing_order->timestamp,
-			"status" => $existing_order->status,
-			"value" => $existing_order->value,
-			"bits" => $existing_order->bits,
-			"bits_payed" => $existing_order->bits_payed
-		);
-
-		return $row_in_array;
-	}
+	/*
+	 * Check for unused address for the currency linked to the order
+	 */		
+    public function getAndUpdateWaitingOrder($orders, $currency, $blockonomics_currency)
+    {
+        foreach ($orders as $order) {
+            //check for currency address already waiting
+            if ($order->blockonomics_currency == $blockonomics_currency && $order->status == -1) {
+                //Check if existing order is expired
+                $current_time = time();
+                $total_time = $this->getTimePeriod() * 60;
+                $clock = $order->timestamp + $total_time - $current_time;
+                if ($clock < 0) {
+                    $order->bits = $this->convertFiatToBlockonomicsCurrency($order->value, $currency, $order->blockonomics_currency);
+                    $order->timestamp = $current_time;
+                }
```

Comment:
Always refresh with latest timestamp. Don't worry about expiry

Reply:
Updated in b5f5be1